### PR TITLE
feat: add missing admin_* methods

### PR DESCRIPTION
**Pull request by alexfertel**
## Motivation

Adds missing `admin_*` RPC methods. Work towards completing #926. 

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by alexfertel** [comment]
Interestingly, the `wasm-wasi` check also fails on `main`.

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, this is a subscription for

https://github.com/ethereum/go-ethereum/blob/473ee8fc07a3f89cf3978e164a6fad218de9a6b5/p2p/peer.go#L92-L103

and it looks like we're missing this type in the admin crate

**Comment by DaniPopes** [comment]
> 
Interestingly, the wasm-wasi check also fails on main. Unsure why ext::anvil::tests::test_anvil_set_interval_mining fails on windows. Damn, why is this PR breaking this stuff...

Not your fault, fixed on main

**Comment by mattsse** [comment]
@alexfertel 

see also 

https://github.com/alloy-rs/alloy/blob/1807965627d29a3e6d9cab7774da1d92be971df5/crates/rpc-client/src/call.rs#L202-L208

"eth_subscribe" is detected as subscription by default, for others we'd need to set it after calling ".request("admin_peeEvents").set_is_subscription"

**Comment by alexfertel** [comment]
@mattsse Does this make more sense?

I'm worried that `PeerEvent` may not serialize/deserialize correctly, is there a place where I can find a few json examples of the responses so that I can test this?

File: crates/provider/src/ext/admin.rs [hunk 1]
```diff
@@ -31,6 +31,17 @@ pub trait AdminApi<N, T>: Send + Sync {
     /// Returns general information about the node as well as information about the running p2p
     /// protocols (e.g. `eth`, `snap`).
     async fn node_info(&self) -> TransportResult<NodeInfo>;
+
+    /// Subscribe to events received by peers over the network.
+    ///
+    /// Like other subscription methods, this returns the ID of the subscription, which is then used
+    /// in all events subsequently.
+    ///
+    /// To unsubscribe from peer events, call `unsubscribe_peer_events`.
```
Comment (DaniPopes) [inline crates/provider/src/ext/admin.rs:40]: doc is outdated

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**